lib/alp_workloads/kvm_workload_utils.pm

```diff
@@ -0,0 +1,201 @@
+    save_screenshot;
+    record_info("kvm-container-manage.sh has been changed to share sol console needed device with kvm container.");
+    assert_script_run("kvm-container-manage.sh create");
+    save_screenshot;
+    record_info("KVM container is created!");
+}
+
+sub cleanup_host_and_kvm_container {
+    assert_script_run("if $ct_engine ps | grep libvirtd; then kvm-container-manage.sh stop;fi");
+    assert_script_run("if $ct_engine ps --all | grep libvirtd; then kvm-container-manage.sh rm;fi");
+    assert_script_run("if which kvm-container-manage.sh;then kvm-container-manage.sh uninstall;fi");
+    save_screenshot;
+    record_info("KVM container is removed and container related files are removed from host");
+}
+
+sub start_kvm_container {
+    assert_script_run("$ct_engine start libvirtd");
+    save_screenshot;
+    assert_script_run("$ct_engine ps | grep -i libvirtd | grep Up");
+    save_screenshot;
+}
+
+sub enter_kvm_container_sh {
+    enter_cmd("clear;$ct_engine exec -ti libvirtd bash", 15);    # interactive mode, so no return immediately
+    save_screenshot;
+    assert_screen('in-libvirtd-container-bash');
+    assert_script_run("echo Hello from libvirtd container");
+    save_screenshot;
+
+    assert_script_run("ip a");
+    save_screenshot;
+    record_info('Please check if it is from within libvirtd container');
+
+}
+
+sub exit_kvm_container {
+    assert_script_run("clear");
+    assert_screen('in-libvirtd-container-bash');
+    enter_cmd("exit");
+    wait_still_screen 3;
+    save_screenshot;
+    assert_screen('back-to-host-shell');
+
+}
+
+sub install_tools_within_kvm_container {
+    my $_tools_to_install = shift;
+
+    $_tools_to_install = "curl openssh hostname gawk supportutils" if (!$_tools_to_install);
+
+    assert_script_run("clear");
+    assert_screen('in-libvirtd-container-bash');
+    zypper_call("in $_tools_to_install", 120);
+    save_screenshot;
+    record_info("Successfully installed test needed tools within VT container.");
+
+}
+
+sub setup_services_within_kvm_container {
+    assert_script_run("clear");
+    assert_screen('in-libvirtd-container-bash');
+
+    # Stop all pre-start virtual network
+    clean_all_virt_networks;
+    # Setup nat virtual network as needed
+    my $_vnet_loc = "/etc/libvirt/qemu/networks/alp-nat-vnet.xml";
+    assert_script_run("curl " . data_url("virt_autotest/alp-nat-vnet.xml") . " -o $_vnet_loc");
+    assert_script_run("virsh net-define $_vnet_loc");
+    assert_script_run("virsh net-autostart test-virt-net");
+    assert_script_run('virsh net-start test-virt-net');
+    save_screenshot;
+    record_info("Successfully setup nat virtual network test-virt-net for automation test.", script_output("ip a"));
+}
+
+sub setup_kvm_container_from_scratch {
+    pull_kvm_container;
+    config_host_and_kvm_container;
+    start_kvm_container;
+    enter_kvm_container_sh;
```

Comment:
I think the name ```enter_kvm_container``` is enough.

Reply:
I prefer current way -- more explicit I think